### PR TITLE
exception_handling / it-academy / JC1-117-22 / Victor Polonnikov / Done tasks 1-7

### PR DESCRIPTION
**Pull request by VictPol**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [ ] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/JavaUtilLogging02.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.rakovets.course.java.core.practice.exception_handling;
+
+import java.util.logging.Logger;
+
+public class JavaUtilLogging02 {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/JavaUtilLogging02.java:5]: remove it

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/JavaUtilLogging01.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.rakovets.course.java.core.practice.exception_handling;
+
+import java.util.logging.Logger;
+
+public class JavaUtilLogging01 {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/JavaUtilLogging01.java:5]: remove it

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Demo01.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        String name = "Victorlong";
+        int counter = 0;
+        try {
+            if (checkNameLength(name)) {
+                counter++;
+            }
+            System.out.println("Более длинных имен уже " + counter);
+            System.out.println("Введите следующее имя для сравнения его длины с заданным именем");
+        } catch (Exception e) {
+            Logger logger = Logger.getLogger(JavaUtilLogging01.class.getName());
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/Demo01.java:18]: Use logger for current class

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Demo01.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    public static void main(String[] args) {
+        String name = "Victorlong";
+        int counter = 0;
+        try {
+            if (checkNameLength(name)) {
+                counter++;
+            }
+            System.out.println("Более длинных имен уже " + counter);
+            System.out.println("Введите следующее имя для сравнения его длины с заданным именем");
+        } catch (Exception e) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/Demo01.java:17]: catch `NullPointerException`

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Demo02.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+import static com.rakovets.course.java.core.practice.exception_handling.Task02.getSumOfFirstNumbersOfArray;
+
+public class Demo02 {
+    public static void main(String[] args) {
+        int[] array = {1, 2, 3, 4, 5};
+        int num = 6;
+        try {
+            int sum = getSumOfFirstNumbersOfArray(array, num);
+            System.out.printf("сумма первых %d чисел массива составляет %d", num, sum);
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/Demo02.java:13]: remove it from prod code

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task03.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
+package com.rakovets.course.java.core.practice.exception_handling;
+
+import com.rakovets.course.java.core.practice.exception_handling.exceptions.MyExceptionTask03;
+
+public class Task03 {
+    final int MINIMUM_TERM_AFTER_OPEN_ACCOUNT = 2;
+    final int MINIMUM_ACCOUNT_AMOUNT = 100;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task03.java:7]:
```suggestion
    private final int MINIMUM_TERM_AFTER_OPEN_ACCOUNT = 2;
    private final int MINIMUM_ACCOUNT_AMOUNT = 100;
```

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/exceptions/MyExceptionTask03.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
+package com.rakovets.course.java.core.practice.exception_handling.exceptions;
+
+public class MyExceptionTask03 extends Exception {
+        private String debugMessage;
+
+        public MyExceptionTask03(String message) {
+            super(message);
+        }
+
+        public MyExceptionTask03(String message, String debugMessage) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/exceptions/MyExceptionTask03.java:10]: remove not use code

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Demo03.java [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+        Task03 currentAccount = new Task03(300, 2021);
+        int restOfAccountAfterRequest = currentAccount.getAccountAmount();
+        try {
+            restOfAccountAfterRequest = currentAccount.getRestOfAccountAfterRequest(150, 2022);
+        } catch (MyExceptionTask03 e) {
+            Logger logger = Logger.getLogger(JavaUtilLogging01.class.getName());
+            logger.warning(e.toString());
+            logger.warning(e.getDebugMessage());
+        }
+        System.out.printf("Остаток на счете после снятия с составляет: %d", restOfAccountAfterRequest);
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/Demo03.java:18]: remove it form prod code

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Demo03.java [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+public class Demo03 {
+    public static void main(String[] args) {
+        Task03 currentAccount = new Task03(300, 2021);
+        int restOfAccountAfterRequest = currentAccount.getAccountAmount();
+        try {
+            restOfAccountAfterRequest = currentAccount.getRestOfAccountAfterRequest(150, 2022);
+        } catch (MyExceptionTask03 e) {
+            Logger logger = Logger.getLogger(JavaUtilLogging01.class.getName());
+            logger.warning(e.toString());
+            logger.warning(e.getDebugMessage());
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/Demo03.java:16]: Where is stacktrace?

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task04.java [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+public class Task04 {
+    final int MINIMUM_AGE_OF_CLIENT = 18;
+    final int MINIMUM_AMOUNT_TO_OPEN_ACCOUNT = 100;
+
+    private int accountAmount;
+
+    {
+        accountAmount = 0;
+    }
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task04.java:14]: remove it
default value for int: 0

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task04.java [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+    public void openAccount(int amountToOpenAccount, int ageOfClient) throws MyExceptionTask04 {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task04.java:15]:
```suggestion
    public void openAccount(int amountToOpenAccount, int ageOfClient) {
```

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/exceptions/MyExceptionTask04.java [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+    private int debugNumber;
+
+    public MyExceptionTask04(String message) {
+        super(message);
+    }
+
+    public MyExceptionTask04(String message, String debugMessage) {
+        super(message);
+        this.debugMessage = debugMessage;
+    }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/exceptions/MyExceptionTask04.java:14]: remove unused code

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Demo04.java [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+import java.util.logging.Logger;
+
+public class Demo04 {
+    public static void main(String[] args) {
+        Task04 currentAccount = new Task04();
+        try {
+            currentAccount.openAccount(180, 15);
+        } catch (MyExceptionTask04 e) {
+            Logger logger = Logger.getLogger(JavaUtilLogging01.class.getName());
+            logger.warning(e.toString());
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/Demo04.java:14]: Where is stacktrace?

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task07.java [hunk 14]
```diff
@@ -0,0 +1,25 @@
+package com.rakovets.course.java.core.practice.exception_handling;
+
+import com.rakovets.course.java.core.practice.exception_handling.exceptions.MyExeptionTask07;
+
+public class Task07 {
+    private final int MAXIMUM_SIZE_OF_ARRAY = 10;
+    public static void printArray(int sizeOfArray, int startToPrintArray, int endToPrintArray, final int MAXIMUM_SIZE_OF_ARRAY) throws MyExeptionTask07{
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task07.java:7]:
```suggestion
    private final int MAXIMUM_SIZE_OF_ARRAY = 10;
    
    public static void printArray(int sizeOfArray, int startToPrintArray, int endToPrintArray, final int MAXIMUM_SIZE_OF_ARRAY) throws MyExeptionTask07{
```

Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task07.java:7]:
```suggestion
    public void printArray(int sizeOfArray, int startToPrintArray, int endToPrintArray, final int MAXIMUM_SIZE_OF_ARRAY) throws MyExeptionTask07{
```

**Review by rakovets — CHANGES_REQUESTED** [review]
You need to fix and add correct solution

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Demo02.java [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+        } catch (ArrayIndexOutOfBoundsException e) {
+            Logger logger = Logger.getLogger(JavaUtilLogging02.class.getName());
```
[the remark this answers is not in the document]
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/exception_handling/Demo02.java:15]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/JavaUtilLogging02.java (hunk 1 above)
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/exception_handling/JavaUtilLogging02.java:5]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/JavaUtilLogging01.java (hunk 2 above)
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/exception_handling/JavaUtilLogging01.java:5]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Demo01.java (hunk 4 above)
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/exception_handling/Demo01.java:17]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Demo02.java (hunk 5 above)
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/exception_handling/Demo02.java:13]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task03.java (hunk 6 above)
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task03.java:7]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/exceptions/MyExceptionTask03.java (hunk 7 above)
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/exception_handling/exceptions/MyExceptionTask03.java:10]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Demo03.java (hunk 8 above)
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/exception_handling/Demo03.java:18]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task04.java (hunk 10 above)
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task04.java:14]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task04.java (hunk 11 above)
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task04.java:15]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/exceptions/MyExceptionTask04.java (hunk 12 above)
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/exception_handling/exceptions/MyExceptionTask04.java:14]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Demo04.java (hunk 13 above)
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/exception_handling/Demo04.java:14]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task07.java (hunk 14 above)
Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task07.java:7]: fixed

Comment (VictPol) [reply src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task07.java:7]: fixed

**Comment by rakovets** [comment]
outdated

**Pull request closed without merging**